packageJsonAnalyzer/codemodList.ts

```diff
@@ -0,0 +1,275 @@
+import {
+	TreeItemCollapsibleState,
+	commands,
+	window,
+	Event,
+	EventEmitter,
+	workspace,
+	TreeItem,
+	ThemeIcon,
+	Uri,
+} from 'vscode';
+import { readFileSync } from 'fs';
+import { buildHash, debounce } from '../src/utilities';
+import { MessageBus, MessageKind } from '../src/components/messageBus';
+import { watchFileWithPattern } from '../src/fileWatcher';
+import { CodemodHash, PathHash, Path, PackageUpgradeItem } from './types';
+import { CodemodItem } from './codemodItem';
+import { commandList } from './constants';
+import { checkIfCodemodIsAvailable, pathExists } from './utils';
+
+class CodemodTreeProvider {
+	#rootPath: string | null;
+	#messageBus: MessageBus;
+	#codemodItemsMap: Map<CodemodHash | PathHash, CodemodItem | Path> =
+		new Map();
+	readonly #eventEmitter = new EventEmitter<void>();
+	public readonly onDidChangeTreeData: Event<void>;
+
+	constructor(rootPath: string | null, messageBus: MessageBus) {
+		this.#rootPath = rootPath;
+		this.#messageBus = messageBus;
+		this.onDidChangeTreeData = this.#eventEmitter.event;
+		this.getPackageJsonListAndWatch();
+
+		this.#messageBus.subscribe(MessageKind.runCodemod, (message) => {
+			const codemodItemFound = this.getTreeItem(
```

Comment:
should we display a message that if the codemod item has been found, we are aborting the "same" execution instance?

Reply:
what do you mean "aborting the "same" execution instance?"

Reply:
this snippet won't launch another execution of the same command if it's been run, am I right?

Reply:
That is a good point but right now it doesn't have a restriction.